guidance/rust_code/src/graph/conversions.rs

```diff
@@ -0,0 +1,113 @@
+use std::f64::consts::PI;
+
+use super::{GPSPoint, TangentialPoint, Graph, Edge, Vertex};
+use nalgebra::{Vector3, Matrix3};
+
+// Distance from the earth relative to the equator (m)
+const RO: f64 = 6378137.0;
+
+// Ecentricity of the earth (unitless?)
+const E: f64 = 0.0818;
+
+// Front entrance of king
+const ORIGIN: GPSPoint = GPSPoint {
+    lat: 34.6147979,
+    long: -112.4509615,
+    height: 1582.3
+};
+
+pub trait IntoTangential {
+    type Output;
+
+    fn into_tangential(self) -> Self::Output;
+}
+
+impl IntoTangential for GPSPoint {
+
+    type Output = TangentialPoint;
+
+    fn into_tangential(self) -> TangentialPoint {
+
+        // vector from center of the earth to an arbitrary point on the map
+        let r_ek_e = self.to_rad().to_xyz();
+
+        // vector from the center of the earth to the origin of our tangential frame
+        let r_eb_e = ORIGIN.to_rad().to_xyz();
+
+        // Vector that goes from origin to an arbitrary point of the map.
+        // Still being expressed in terms of a rectangular frame with its origin at the center of
+        // the earth.
+        let r_kb_e = r_eb_e - r_ek_e;
+
+        // The transformation matrix that converts the basis or our coordinate system
+        let c_e_k = Matrix3::new(
+            -(self.long.cos() * self.lat.sin()),       -ORIGIN.long.sin(),   -(ORIGIN.lat.cos() * ORIGIN.long.cos()), // Check this line
```

Comment:
I would recommend using self.long.to_radians.cos() instead of hoping that the user converted the GPSPoint to radians before calling this function

Reply:
radian conversion does happen within this function. The user will not have to remember to convert the GPSPoint to radian. In fact, they won't be able to since to_rad is private. The only thing the user will be able to do is to call the into_tangential function which takes the GPSPoint in degrees and converts it to our tangential frame.

Reply:
Hang on. I lied about the conversion to radians of self. I implemented your suggestion. Really good catch. This was 100% a bug.

Reply:
Is there a point other than the origin which would be good to check the to_tangential conversion?

Reply:
Try some points that are some distance true north or true east. True north would be +x, no change y, and a little change z (depends a bit on elevation). True east would be no change x, +y, and a little change z

You may be able to get some distances is meters or feet from a view on google maps, maybe play around with that to check approximate distance changes